src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java

```diff
@@ -711,11 +724,22 @@ private float convertRowToArrow(
           throw new SFException(
               ErrorCode.INVALID_ROW, columnName, "Passed null to non nullable field");
         } else {
-          insertNull(vector, stats, curRowIndex);
+          insertNull(vector, forkedStats, curRowIndex);
         }
       }
     }
 
+    // All input values passed validation, iterate over the columns again and combine their existing
+    // statistics
+    // with the forked statistics for the current row.
+    for (Map.Entry<String, Object> entry : row.entrySet()) {
+      String key = entry.getKey();
+      String columnName = LiteralQuoteUtils.unquoteColumnName(key);
```

Comment:
This will be made more efficient in https://github.com/snowflakedb/snowflake-ingest-java/pull/362.

Reply:
I added another local map to avoid this unnecessary re-computation. When we move to caffeine, we can get rid of it because caffeine cache performance is close to HashMap performance in this case.